cpp/src/arrow/util/basic_decimal.cc

```diff
@@ -1391,4 +1347,8 @@ BasicDecimal256 operator/(const BasicDecimal256& left, const BasicDecimal256& ri
   return result;
 }
 
+// Explicitly instantiate template base class, for DLL linking on Windows
```

Comment:
Does this approach avoid putting all template class implementation code inside the header?

Reply:
It should, though that's not the goal here :-)